extension/experimental/storage/nop_client.go

```diff
@@ -10,6 +10,7 @@ type nopClient struct{}
 var nopClientInstance Client = &nopClient{}
 
 // NewNopClient returns a nop client
+// Deprecated: [0.85.0] Use helpers from github.com/open-telemetry/opentelemetry-collector-contrib/extension/storage/storagetest instead.
```

Comment:
It's used in non-test code [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/789811ca8c728b7684bc8f80cd44cfe9523c2b2a/pkg/stanza/adapter/storage.go#L16).

Personally I find this useful as a default implementation of the interface. Code that would interact with storage _if_ it was specified doesn't need to be concerned with whether or not it actually was.